### PR TITLE
feat: ConversationDAO

### PR DESCRIPTION
**Pull request by typfel**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

Add persistence of conversations and its members

### Solutions

Introduce ConversationDAO

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: persistence/build.gradle.kts [hunk 1]
```diff
@@ -14,6 +14,7 @@ dependencies {
 
 sqldelight {
     database("AppDatabase") {
+        dialect = "sqlite:3.24"
```
Comment (typfel) [inline persistence/build.gradle.kts:17]: Unlock newer sqlite API like: `ON CONFLICT` for doing "upserts"

File: persistence/src/commonMain/kotlin/com/wire/kalium/persistence/dao/ConverationDAOImpl.kt [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+import com.squareup.sqldelight.runtime.coroutines.mapToOneOrNull
+import com.wire.kalium.persistence.db.ConverationsQueries
+import com.wire.kalium.persistence.db.MembersQueries
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+import com.wire.kalium.persistence.db.Conversation as SQLDelightConversation
+import com.wire.kalium.persistence.db.Member as SQLDelightMember
+
+class ConverationMapper {
+    fun toDAO(conversation: SQLDelightConversation): Conversation {
```
Comment (vitorhugods) [inline persistence/src/commonMain/kotlin/com/wire/kalium/persistence/dao/ConverationDAOImpl.kt:14]: Shouldn't this be something like `toModel` or `fromDatabaseRecord` or `fromDbEntry`, something like that? It doesn't actually map to a `DAO`.

File: persistence/src/commonMain/kotlin/com/wire/kalium/persistence/dao/ConverationDAOImpl.kt [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        return Conversation(conversation.qualified_id, conversation.name)
+    }
+}
+
+class MemberMapper {
+    fun toDAO(member: SQLDelightMember): Member {
```
Comment (vitorhugods) [inline persistence/src/commonMain/kotlin/com/wire/kalium/persistence/dao/ConverationDAOImpl.kt:20]: Same here about naming.

File: persistence/src/commonMain/kotlin/com/wire/kalium/persistence/dao/ConverationDAOImpl.kt [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+import com.squareup.sqldelight.runtime.coroutines.mapToList
+import com.squareup.sqldelight.runtime.coroutines.mapToOneOrNull
+import com.wire.kalium.persistence.db.ConverationsQueries
+import com.wire.kalium.persistence.db.MembersQueries
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+import com.wire.kalium.persistence.db.Conversation as SQLDelightConversation
+import com.wire.kalium.persistence.db.Member as SQLDelightMember
+
+class ConverationMapper {
```
Comment (vitorhugods) [inline persistence/src/commonMain/kotlin/com/wire/kalium/persistence/dao/ConverationDAOImpl.kt:13]:
```suggestion
class ConversationMapper {
```

File: persistence/src/commonMain/kotlin/com/wire/kalium/persistence/dao/ConverationDAOImpl.kt [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+    fun toDAO(member: SQLDelightMember): Member {
+        return Member(member.user)
+    }
+}
+
+class ConversationDAOImpl(private val conversationQueries: ConverationsQueries,
+                          private val memberQueries: MembersQueries): ConversationDAO {
+
+    val memberMapper = MemberMapper()
+    val conversationMapper = ConverationMapper()
```
Comment (vitorhugods) [inline persistence/src/commonMain/kotlin/com/wire/kalium/persistence/dao/ConverationDAOImpl.kt:29]:
```suggestion
  val conversationMapper = ConversationMapper()
```

**Review by vitorhugods — CHANGES_REQUESTED** [review]
Again: 🧑‍🍳🤌🏻

Just small comment about naming and a typo on `Converation/Conversation`

**Review by vitorhugods — APPROVED** [review]

**Pull request merged**